Tests/Common/Util/FactorFileGeneratorTests.cs

```diff
@@ -80,6 +79,16 @@ public void FactorFiles_CanBeGenerated_Accurately()
 
             var originalFactorFileInstance = FactorFile.Read(PermTick, Market);
 
+            // start date is the day of the first dividend/split entry in the table
+            var start = originalFactorFileInstance.SortedFactorFileData.Keys.Skip(1).First();
```

Comment:
This is a great change. Instead of picking some arbitrary date in the past (unix epoch), we pick the first date as provided in the reference files. This is OK since the test already depends on the reference files and we wouldn't want to produce factor before or after (_especially_ after) what we are able to compare against.